jena-base/src/main/java/org/apache/jena/atlas/iterator/Iter.java

```diff
@@ -29,6 +29,14 @@
 import org.apache.jena.atlas.lib.Closeable ;
 import org.apache.jena.atlas.lib.Sink ;
 
+/**
+ * Iter provides utilities for working with {@link Iterator}s.
+ *
+ * Iter should never be used as a return type or parameter type in the public contract of a class. It is only to be used
+ * inside implementation code and is instantiated only to allow method-chaining as part of a calculation.
```

Comment:
This is exactly what I needed. Sorry to have bugged you so much about it, but I really think this deserves to be written down. I will correct this PR to advise just what you just did.

I am a little surprised that Jena's policy is that once a type is exposed, it can never be changed. That seems extraordinarily restrictive. But that's a completely different question.